### PR TITLE
port support for client vpn endpoint

### LINKED ISSUE #11633: AWS Client VPN now supports port configuration
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

You can now configure your AWS Client VPN endpoint to use either the port 443 or the port 1194.

### New or Affected Resource(s)

* [`aws_ec2_client_vpn_endpoint`](https://www.terraform.io/docs/providers/aws/r/ec2_client_vpn_endpoint.html)

### Potential Terraform Configuration

```hcl
resource "aws_ec2_client_vpn_endpoint" "example" {
  vpn_port = 1194
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/01/aws-client-vpn-now-supports-port-configuration/).
[API reference](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_ClientVpnEndpoint.html).

Requires [AWS SDK v1.28.4](https://github.com/aws/aws-sdk-go/releases/tag/v1.28.4):
* https://github.com/terraform-providers/terraform-provider-aws/pull/11630

### PR DESCRIPTION
**Pull request by ranga543**
Changes proposed in this pull request:
* Added `vpn_port` to `aws_ec2_client_vpn_endpoint` resource.

Output from acceptance testing:
```
make testacc TESTARGS='-run=TestAccAwsEc2ClientVpnEndpoint_vpnPort'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -count 1 -parallel 20 -run=TestAccAwsEc2ClientVpnEndpoint_vpnPort -timeout 120m
?       github.com/terraform-providers/terraform-provider-aws   [no test files]
=== RUN   TestAccAwsEc2ClientVpnEndpoint_vpnPort
=== PAUSE TestAccAwsEc2ClientVpnEndpoint_vpnPort
=== CONT  TestAccAwsEc2ClientVpnEndpoint_vpnPort
--- PASS: TestAccAwsEc2ClientVpnEndpoint_vpnPort (41.28s)
```
Closes #11633

### CODE REVIEW

**Comment by e-moshaya** [comment]
can this be merged please?

**Comment by DrFaust92** [comment]
Closes #11633

File: aws/resource_aws_ec2_client_vpn_endpoint_test.go [hunk 1]
```diff
@@ -275,6 +275,38 @@ func TestAccAwsEc2ClientVpnEndpoint_splitTunnel(t *testing.T) {
 	})
 }
 
+func TestAccAwsEc2ClientVpnEndpoint_vpnPort(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (ranga543) [reply aws/resource_aws_ec2_client_vpn_endpoint_test.go]: Updated this one as suggested.

File: aws/resource_aws_ec2_client_vpn_endpoint_test.go [hunk 2]
```diff
@@ -275,6 +275,38 @@ func TestAccAwsEc2ClientVpnEndpoint_splitTunnel(t *testing.T) {
...
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_ec2_client_vpn_endpoint.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
```
[the remark this answers is not in the document]
Comment (ranga543) [reply aws/resource_aws_ec2_client_vpn_endpoint_test.go]: @gdavison  Updated this one as suggested.

File: aws/resource_aws_ec2_client_vpn_endpoint_test.go [hunk 3]
```diff
@@ -523,3 +555,42 @@ resource "aws_ec2_client_vpn_endpoint" "test" {
...
+    type                       = "certificate-authentication"
+    root_certificate_chain_arn = "${aws_acm_certificate.test.arn}"
+  }
+
+  connection_log_options {
+    enabled = false
+  }
+}
+`, rName)
+}
```
[the remark this answers is not in the document]
Comment (ranga543) [reply aws/resource_aws_ec2_client_vpn_endpoint_test.go]: Updated this one as suggested.

File: website/docs/r/ec2_client_vpn_endpoint.html.markdown [hunk 4]
```diff
@@ -45,6 +45,7 @@ The following arguments are supported:
 * `split_tunnel` - (Optional) Indicates whether split-tunnel is enabled on VPN endpoint. Default value is `false`.
 * `tags` - (Optional) A mapping of tags to assign to the resource.
 * `transport_protocol` - (Optional) The transport protocol to be used by the VPN session. Default value is `udp`.
+* `vpn_port` - (Optional) The port number for the Client VPN endpoint. Default value is `443`.
```
[the remark this answers is not in the document]
Comment (ranga543) [reply website/docs/r/ec2_client_vpn_endpoint.html.markdown]: Updated this one as suggested.

**Comment by ranga543** [comment]
@gdavison any status ?

**Comment by DrFaust92** [comment]
Hey @ranga543 can you add `Closes #11633` to you PR description. this way it will link the PR to the issue

**Comment by ranga543** [comment]
@DrFaust92 just updated PR description to include closes.

**Comment by ranga543** [comment]
@DrFaust92 any status ?

**Comment by russmac** [comment]
@ranga543 Firstly thanks for your work on this :100: Could you resolve the outstanding question? (It appears its resolved just not clicked resolved) https://github.com/hashicorp/terraform-provider-aws/pull/12104#discussion_r453233907

It might be causing this PR to be filtered out by Hashicorp. :+1:

**Comment by ranga543** [comment]
@russmac done.

**Comment by russmac** [comment]
@gdavison  Any requests for the PR author?

**Comment by ewbankkit** [comment]
Superseded by #22793.

**Comment by ewbankkit** [comment]
@ranga543 Thanks for the contribution :tada: :clap:.
I have incorporated your changes into #22793.

**Pull request closed without merging**